### ISSUE TITLE
Problems with seneca.ready firing in 2.x

### ISSUE THREAD

**Issue by mcdonnelldean** [senecajs/seneca#387]
I've spent a whole day wasted tracking issues to ready not firing. Should it be firing? Is it depreciated or should it be?

**Comment by mirceaalexandru** [comment]
Maybe this is also related:

https://github.com/senecajs/seneca-web/issues/59

seems that this is not called in seneca:
https://github.com/senecajs/seneca/blob/master/seneca.js#L949


@geek here is an example:

```
seneca.use(require('seneca-cron'))

seneca.ready(function(){
    console.log('READY CALLED')
    seneca.use(require('seneca-user'))
})
```

with seneca 2.0 ready is never called, with 1.4.0 is called correctly

**Comment by mcdonnelldean** [comment]
Related to #398

**Comment by mcdonnelldean** [comment]
@mirceaalexandru When you get back can you write a test for this so we can fix it? Add it to the existing tests and skip it on PR so it doesn't break the build. I'll pull it and use it to fix the issue.

**Comment by mihaidma** [comment]
I also hit this issue: https://github.com/senecajs/seneca-postgres-store/issues/47

**Comment by mcollina** [comment]
Some of the work I'm doing in #402 will fix it 🎉 , or possibly made it worse 😢 .

I can say the current implementation is broken.

**Comment by mihaidma** [comment]
:fireworks:

**Comment by mcdonnelldean** [comment]
This is not a bug, in the end it is an ordering issue. I am documenting the process as it should be and updating the site with this info. Once done I will close the issue. @mirceaalexandru @mihaidma

**Comment by mcdonnelldean** [comment]
Merged with #413 To them all together.
